airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/test/steps/java_connectors.py

```diff
@@ -168,7 +163,6 @@ def get_test_steps(context: ConnectorContext) -> STEP_TREE:
         normalization_steps = _get_normalization_steps(context)
         steps.append(normalization_steps)
 
-    acceptance_test_steps = _get_acceptance_test_steps(context)
-    steps.append(acceptance_test_steps)
+    steps.append(optional_test_suites_to_run)
```

Comment:
❗ I think this shows a case that were not supporting

1. We want to remove a subset of tests from the step tree but those steps can occur in any location in the tree

A better approach is to either
**1. Have a filter function you can apply to the tree that removes steps by id and outputs a new tree**
```py
def remove_steps(step_tree: STEP_TREE, remove_steps: List[str]) -> STEP_TREE:
    # ...

filtered_steps = remove_steps(step_tree, steps_to_remove)
run_steps(filtered_steps)
```

**2. Make use of our skip step functionallity in run_steps**
https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/pipelines/helpers/execution/run_steps.py#L81


**I think #2 is the best approach**

Reply:
> ❗ I think this shows a case that were not supporting
> 
> 1. We want to remove a subset of tests from the step tree but those steps can occur in any location in the tree
> 
> A better approach is to either **1. Have a filter function you can apply to the tree that removes steps by id and outputs a new tree**
> 
> ```python
> def remove_steps(step_tree: STEP_TREE, remove_steps: List[str]) -> STEP_TREE:
>     # ...
> 
> filtered_steps = remove_steps(step_tree, steps_to_remove)
> run_steps(filtered_steps)
> ```
> 
> **2. Make use of our skip step functionallity in run_steps** https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/pipelines/helpers/execution/run_steps.py#L81
> 
> **I think #2 is the best approach**

@bnchrch fair point!
I changed the logic to leverage `RunStepsOptions`. It's definitely cleaner as:
* It allows us to not alter the concurrency settings we declare while building the STEP TREE
* Skipped tests are still rendered in the report so user can be aware that these step could run if they're enabled.